src/scalar_planner_estimates.rs

```diff
@@ -0,0 +1,220 @@
+    BUTTERFLY_SLOPE_19 * repeats as f32 + BUTTERFLY_CONST_19
+}
+
+const BUTTERFLY_SLOPE_23: f32 = 41.35303;
+const BUTTERFLY_CONST_23: f32 = -1.76352;
+
+pub fn estimate_butterfly_cost_23(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_23 * repeats as f32 + BUTTERFLY_CONST_23
+}
+
+const BUTTERFLY_SLOPE_29: f32 = 84.98456;
+const BUTTERFLY_CONST_29: f32 = -1.58332;
+
+pub fn estimate_butterfly_cost_29(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_29 * repeats as f32 + BUTTERFLY_CONST_29
+}
+
+const BUTTERFLY_SLOPE_31: f32 = 100.00000;
+const BUTTERFLY_CONST_31: f32 = -4.88995;
+
+pub fn estimate_butterfly_cost_31(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_31 * repeats as f32 + BUTTERFLY_CONST_31
+}
+
+const BUTTERFLY_SLOPE_32: f32 = 36.36041;
+const BUTTERFLY_CONST_32: f32 = -0.08669;
+
+pub fn estimate_butterfly_cost_32(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_32 * repeats as f32 + BUTTERFLY_CONST_32
+}
+
+// --- End code generated by tools/estimates/fit_butterflies.py ---
+
+// --- Begin code generated by tools/estimates/fit_radix4.py ---
+
+const RADIX4_CONST: f32 = 0.90900;
+const RADIX4_SLOPE: f32 = 0.58761;
+const RADIX4_EXP: f32 = 1.15980;
+
+pub fn estimate_radix4_cost(len: usize, repeats: usize) -> f32 {
+    (RADIX4_CONST + RADIX4_SLOPE * (len as f32).powf(RADIX4_EXP)) * repeats as f32
+}
+
+// --- End code generated by tools/estimates/fit_radix4.py ---
+
+// --- Begin code generated by tools/estimates/fit_raders.py ---
+
+const RADERS_CONST: f32 = -1.22742;
+const RADERS_SLOPE: f32 = 2.30372;
+
+pub fn estimate_raders_cost(inner_fft: &Arc<Recipe>, repeats: usize) -> f32 {
+    (RADERS_CONST + RADERS_SLOPE * (inner_fft.len() + 1) as f32 + 2.0 * inner_fft.cost(1))
+        * repeats as f32
+}
+
+// --- End code generated by tools/estimates/fit_raders.py ---
+
+// --- Begin code generated by tools/estimates/fit_bluesteins.py ---
+
+const BLUESTEINS_CONST: f32 = 16.77562;
+const BLUESTEINS_LEN_SLOPE: f32 = 0.30714;
+const BLUESTEINS_INNER_LEN_SLOPE: f32 = 0.34986;
+
+pub fn estimate_bluesteins_cost(len: usize, inner_fft: &Arc<Recipe>, repeats: usize) -> f32 {
+    (BLUESTEINS_CONST
+        + BLUESTEINS_INNER_LEN_SLOPE * inner_fft.len() as f32
+        + BLUESTEINS_LEN_SLOPE * len as f32
+        + 2.0 * inner_fft.cost(1))
+        * repeats as f32
+}
+
+// --- End code generated by tools/estimates/fit_bluesteins.py ---
+
+// --- Begin code generated by tools/estimates/fit_mixedradixes.py ---
+
+const MIXEDRADIX_SLOPE: f32 = 0.09496;
+const MIXEDRADIX_EXP: f32 = 1.18301;
+
+pub fn estimate_mixedradix_cost(
+    len: usize,
+    left_fft: &Arc<Recipe>,
+    right_fft: &Arc<Recipe>,
+    repeats: usize,
+) -> f32 {
+    let inners_cost = left_fft.cost(right_fft.len()) + right_fft.cost(left_fft.len());
+    let twiddle_cost = MIXEDRADIX_SLOPE * (len as f32).powf(MIXEDRADIX_EXP);
+    (inners_cost + twiddle_cost) * repeats as f32
+}
+
+const MIXEDRADIXSMALL_SLOPE: f32 = 1.07425;
+const MIXEDRADIXSMALL_EXP: f32 = 0.86855;
```

Comment:
This exponent being smaller than the Mixed Radix exponent suggests to me that 1: we don't have enough data for this algorithm, and B: it might be worth testing it and using it at larger sizes than just 31x31. That doesn't need to be addressed in this PR, but might be worth looking at in the future.

Reply:
I cheat a little here..
I started by measuring MixedRadix vs MixedRadixSmall (and same for GoodThomas). At short lengths, the Small versions win because of less overhead. And at long lenghts, the Small versions get a big slowdown (due to cache misses I assume). Then in between, at lengths from a few hundred to 100k or so, they are so similar I can't reliably decide which is faster.
So I went with a simpler approach, by stiching them together. The Small versions are used when both inners are <= 32, and are only fitted in that range. The general ones are used for anything above that, and only fitted above that.